### ISSUE TITLE
Add support to chainable validations

### ISSUE THREAD

**Issue by likern** [samuelcolvin/pydantic#472]
# Question | Feature Request | Bug

For bugs/questions:
* OS: Ubuntu 18.04
* Python version: 3.7.3 (default, Mar 26 2019, 00:55:50)  [GCC 7.3.0]
* Pydantic version: 0.23

**Note**: I [asked this question](https://stackoverflow.com/questions/55666792/how-to-chain-validations-with-pydantic) on StackOverflow.

Now, BaseModel objects can only be constructed from dict. But it's very desirable to construct BaseModel object from another BaseModel objects.

This is my case:
I have a webhook which is used for whole protocol. All different data types are passed as json body.
```py
@app.route("/", methods=['POST'])
async def telegram_webhook(request):
    update = Update.parse_obj(request.json)
    /* do something with update */
```
I use Update model for validation (which is used are base validation):
```py
class Update(BaseModel):
    update_id: int
    message: Optional[Message]
    ...

class Message(BaseModel):
    message_id: int
    text: Optional[str]
```
But later, I want to apply additional validation to Message (so to check that it is TextMessage and to convert to it):
```py
class TextMessage(Message):
    text: str # Now is required

    @validator('text')
    def check_text_length(cls, value):
        length = len(value)
        if length > 4096:
            raise ValueError(f'text length {length} is too large')
        return value
```

So I get Message object in function and get expection:
```
def process_text_message(message):
    text_message = TextMessage.parse_obj(message)
```

How to solve this?
It would be great, if I could get entry dict, convert it to BaseModel object and process all validation / conversions / etc through BaseModel, without conversions to dict. And at the end, as last step, to convert it back to dict (for example to insert into MongoDB)

**Comment by samuelcolvin** [comment]
I've given a quick and dirty answer on SO: use `parse_obj(model.dict())`.

But the real solution is to improve `parse_obj` to cope with dict-like objects.

Something like 

```py
        if not isinstance(obj, dict):
            exc = TypeError(f'{cls.__name__} expected dict not {type(obj).__name__}')
            raise ValidationError([ErrorWrapper(exc, loc='__obj__')])
        return cls(**obj)
```

Should be replaced with

```py
try:
    cls(**dict(obj))
except (KeyError, ValueError) as e:
    raise ValidationError([ErrorWrapper(exc, loc='__obj__')]) from e
```

**Comment by samuelcolvin** [comment]
should be fixed in #476

**Comment by likern** [comment]
@samuelcolvin Thank you very much for answer and great support. I'll be eagerly await fix in new release.
